libs/domains/services/data-access/src/lib/domains-services-data-access.ts

```diff
@@ -812,6 +841,100 @@ export const mutations = {
     const response = await mutation()
     return response.data
   },
+  async createCustomDomain({
+    serviceId,
+    serviceType,
+    payload,
+  }: {
+    serviceId: string
+    serviceType: Extract<ServiceType, 'APPLICATION' | 'CONTAINER'>
+    payload: CustomDomainRequest
+  }) {
+    const { mutation } = match(serviceType)
+      .with('APPLICATION', () => ({
+        mutation: customDomainApplicationApi.createApplicationCustomDomain.bind(
+          customDomainApplicationApi,
+          serviceId,
+          payload
+        ),
+        serviceType,
+      }))
+      .with('CONTAINER', () => ({
+        mutation: customDomainContainerApi.createContainerCustomDomain.bind(
+          customDomainContainerApi,
+          serviceId,
+          payload
+        ),
+        serviceType,
+      }))
+      .exhaustive()
+    const response = await mutation()
+    return response.data
+  },
+  async editCustomDomain({
+    serviceId,
+    serviceType,
+    customDomainId,
+    payload,
+  }: {
+    serviceId: string
+    serviceType: Extract<ServiceType, 'APPLICATION' | 'CONTAINER'>
+    customDomainId: string
+    payload: CustomDomainRequest
+  }) {
+    const { mutation } = match(serviceType)
+      .with('APPLICATION', () => ({
+        mutation: customDomainApplicationApi.editCustomDomain.bind(
+          customDomainApplicationApi,
+          serviceId,
+          customDomainId,
+          payload
+        ),
+        serviceType,
+      }))
+      .with('CONTAINER', () => ({
+        mutation: customDomainContainerApi.editContainerCustomDomain.bind(
+          customDomainContainerApi,
+          serviceId,
+          customDomainId,
+          payload
+        ),
+        serviceType,
+      }))
+      .exhaustive()
+    const response = await mutation()
+    return response.data
+  },
+  async deleteCustomDomain({
+    serviceId,
+    serviceType,
+    customDomainId,
+  }: {
+    serviceId: string
+    serviceType: Extract<ServiceType, 'APPLICATION' | 'CONTAINER'>
+    customDomainId: string
+  }) {
+    const { mutation } = match(serviceType)
+      .with('APPLICATION', (serviceType) => ({
+        mutation: customDomainApplicationApi.deleteCustomDomain.bind(
+          applicationMainCallsApi,
+          serviceId,
+          customDomainId
+        ),
+        serviceType,
+      }))
+      .with('CONTAINER', (serviceType) => ({
+        mutation: customDomainContainerApi.deleteContainerCustomDomain.bind(
+          containerMainCallsApi,
+          serviceId,
+          customDomainId
+        ),
+        serviceType,
+      }))
+      .exhaustive()
+    const response = await mutation()
```

Comment:
You can pass `serviceId` and `customDomainId` here because they are not type dependent